controller/script/simulate-proxy/main.go

```diff
@@ -22,7 +22,99 @@ import (
 	_ "k8s.io/client-go/plugin/pkg/client/auth"
 )
 
-/* A simple script for posting simulated telemetry data to the proxy api */
+/* A simple script for exposing simulated prometheus metrics */
+
+type simulatedProxy struct {
+	sleep time.Duration
+}
+
+type proxyVectors struct {
+	requestTotals      *prom.CounterVec
+	responseTotals     *prom.CounterVec
+	responseDurationMs *prom.HistogramVec
+	responseLatencyMs  *prom.HistogramVec
+}
+
+// generateProxyTraffic randomly creates metrics under the guise of a single conduit proxy routing traffic.
+func (s *simulatedProxy) generateProxyTraffic(namespace string, podList []*v1.Pod, proxyVec *proxyVectors) {
+	inboundPod := podList[rand.Intn(len(podList))]
+
+	ip := randomPodIp(podList, nil)
+
+	for {
+		inboundRandomCount := int(rand.Float64() * 10)
+		outboundRandomCount := int(rand.Float64() * 10)
+		randomOutboundPod := podList[rand.Intn(len(podList))]
+
+		proxyVec.
+			requestTotals.
+			With(overrideDefaultLabels(prom.Labels{
+				"direction":  "inbound",
+				"deployment": strings.Split(inboundPod.GetName(), "-")[0],
+				"namespace":  namespace,
+				"instance":   ip.String(),
+				"authority":  "world.greeting:7778",
+			})).
+			Add(float64(inboundRandomCount))
+
+		proxyVec.
+			responseTotals.
+			With(overrideDefaultLabels(prom.Labels{
+				"direction":   "outbound",
+				"deployment":  strings.Split(inboundPod.GetName(), "-")[0],
+				"namespace":   namespace,
+				"instance":    ip.String(),
+				"authority":   "world.greeting:7778",
+				"status_code": fmt.Sprintf("%d", randomHttpResponseCode()),
+			})).
+			Add(float64(outboundRandomCount))
+
+		observeLatencies(
+			randomLatencies(randomCount()),
+			proxyVec.responseLatencyMs,
+			overrideDefaultLabels(prom.Labels{
+				"direction":      "outbound",
+				"dst_deployment": strings.Split(randomOutboundPod.GetName(), "-")[0],
+				"status_code":    strconv.Itoa(http.StatusOK),
+				"namespace":      namespace,
+				"authority":      "world.greeting:7778",
+			}))
+
+		observeLatencies(
+			randomLatencies(randomCount()),
+			proxyVec.responseLatencyMs,
+			overrideDefaultLabels(prom.Labels{
+				"direction":        "outbound",
+				"dst_deployment":   strings.Split(randomOutboundPod.GetName(), "-")[0],
+				"grpc_status_code": strconv.Itoa(int(randomGrpcResponseCode())),
+				"namespace":        namespace,
+				"authority":        "world.greeting:7778",
+			}))
+
+		observeLatencies(
+			randomLatencies(randomCount()),
+			proxyVec.responseDurationMs,
+			overrideDefaultLabels(prom.Labels{
+				"direction":        "outbound",
+				"dst_deployment":   strings.Split(randomOutboundPod.GetName(), "-")[0],
+				"grpc_status_code": strconv.Itoa(int(randomGrpcResponseCode())),
+				"namespace":        namespace,
+				"authority":        "world.greeting:7778",
+			}))
+
+		time.Sleep(s.sleep)
+	}
+}
+func observeLatencies(latencyBuckets []uint32, responseLatencies *prom.HistogramVec, latencyLabels prom.Labels) {
```

Comment:
as this is not specific to responses, consider renaming `responseLatencies` to `latencies`?